### PR TITLE
Improved JDK 22-23 compatibility, reproducible builds support

### PR DESCRIPTION
**Pull request by wendigo**
PMD 7.0.0 is finally compatible with JDK 21

### CODE REVIEW

**Comment by wendigo** [comment]
cc @electrum

**Comment by wendigo** [comment]
@martint @electrum PTAL

**Comment by wendigo** [comment]
Rebased, resolved conflicts and added maven-gpg-plugin

**Comment by wendigo** [comment]
@martint @electrum ptal

**Comment by wendigo** [comment]
@martint @electrum gentle reminder

**Comment by wendigo** [comment]
Ping @martint @electrum

**Comment by wendigo** [comment]
@martint @electrum PTAL

File: airbase/pom.xml [hunk 1]
```diff
@@ -1273,7 +1273,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-gpg-plugin</artifactId>
-                    <version>3.2.1</version>
+                    <version>3.2.2</version>
```
Comment (electrum) [inline airbase/pom.xml]: This should be 3.2.4

File: CHANGES.md [hunk 2]
```diff
@@ -12,7 +12,8 @@ Airbase 155
  - spotbugs-annotations 4.8.4 (from 4.8.3)
  - jakarta.annotation-api 3.0.0 (from 2.1.1)
  - opentelemetry 1.37.0 (from 1.36.0)
-
```
Comment (electrum) [inline CHANGES.md:4]: Something weird happened with the diff here. No action needed, but I'm puzzled.

File: CHANGES.md [hunk 3]
```diff
@@ -14,7 +14,8 @@ Airbase 155
  - jakarta.annotation-api 3.0.0 (from 2.1.1)
  - opentelemetry 1.37.0 (from 1.36.0)
  - logback 1.5.6 (from 1.5.3)
-
```
Comment (electrum) [inline CHANGES.md]: Same thing here

File: airbase/pom.xml [hunk 4]
```diff
@@ -910,11 +912,11 @@
                     <configuration>
                         <!-- Include only properties used above to speed up build (https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/462) -->
                         <includeOnlyProperties>
-                            <includeOnlyProperty>\Qgit.build.time</includeOnlyProperty>
                             <includeOnlyProperty>\Qgit.commit.id</includeOnlyProperty>
+                            <includeOnlyProperty>\Qgit.commit.time</includeOnlyProperty>
```
Comment (electrum) [inline airbase/pom.xml:950]: Move this after the next line so that properties are sorted?

**Review by electrum — APPROVED** [review]
This is great! Thanks for doing this.

**Comment by wendigo** [comment]
@electrum I'll post another PR where I start with the [Move all versions to properties](https://github.com/airlift/airbase/pull/403/commits/6f75b7fb95d4775d8c338c862d0682311848ab57) as a first commit

**Comment by wendigo** [comment]
@electrum here's the new PR: https://github.com/airlift/airbase/pull/407

**Pull request closed without merging**